app/main.py

```diff
@@ -95,16 +104,68 @@ def _decrypt(self, encrypted_jwt, tx_id):
             raise RetryableError()
 
     def run(self):
+        logger.debug("Starting consumer")
         self.consumer.run()
 
     def stop(self):
+        logger.debug("Stopping consumer ")
         self.consumer.stop()
 
 
+class HealthCheck(tornado.web.RequestHandler):
+
+    @staticmethod
+    def rabbit_health():
+        http_client = HTTPClient()
+        resp = http_client.fetch(settings.RABBIT_HEALTHCHECK_URL, auth_username=settings.SEFT_RABBITMQ_MONITORING_USER,
+                                 auth_password=settings.SEFT_RABBITMQ_MONITORING_PASS)
+        body = resp.body.decode('utf8')
+        body = json.loads(body)
+        if body.get('status') == "ok":
+            health = body.get('status')
+        else:
+            health = "failed"
+        return health
+
+    @staticmethod
+    def ftp_health():
+        ftp = SDXFTP(logger, settings.FTP_HOST, settings.FTP_USER, settings.FTP_PASS, settings.FTP_PORT)
+        try:
+            ftp.get_connection()
+            health = "ok"
+        except Error as e:
+            logger.debug("FTP error raised" + e)
+            health = "failed"
+        return health
+
+    def initialize(self):
+        self.get_health()
+        threading.Timer(300, self.get_health()).start()
+
+    def get_health(self):
+        self.rabbit_status = self.rabbit_health()
+        self.ftp_status = self.ftp_health()
+        if self.rabbit_status == "ok" and self.ftp_status == "ok":
+            self.app_status = "ok"
+        else:
+            self.app_status = "failed"
+
+    def get(self):
+        self.write({"status": self.app_status, "dependencies": {"rabbitmq": self.rabbit_status, "ftp": self.ftp_status}})
+
+
+def make_app():
+    return tornado.web.Application([
+        (r"/healthcheck", HealthCheck),
+    ])
+
+
 def main():
     logger.debug("Starting SEFT consumer service")
 
     seft_consumer = SeftConsumer()
+    app = make_app()
+    app.listen(8080)
```

Comment:
Port should be configurable using a `PORT` environment var (CF/other PaaS compatible then)